src/core/thread/dua_manager.cpp

```diff
@@ -184,6 +249,522 @@ otError DuaManager::Store(void)
+            {
+                uint16_t childIndex = Get<ChildTable>().GetChildIndex(child);
+
+                if (mChildDuaMask.Get(childIndex) && !mChildDuaRegisterMask.Get(childIndex))
+                {
+                    mChildIndexDuaRegistering = childIndex;
+                }
+            }
+        }
+        {
+            uint32_t lastTransactionTime;
+            Child *  child = Get<ChildTable>().GetChildAtIndex(mChildIndexDuaRegistering);
+
+            OT_ASSERT(child->GetDomainUnicastAddress(dua) == OT_ERROR_NONE);
+
+            lastTransactionTime = Time::MsecToSec(TimerMilli::GetNow() - child->GetLastHeard());
+
+            SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kTarget, &dua, sizeof(dua)));
+            SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kMeshLocalEid, &child->GetMeshLocalIid(),
+                                                 sizeof(Ip6::InterfaceIdentifier)));
+            SuccessOrExit(error = Tlv::AppendUint32Tlv(*message, ThreadTlv::kLastTransactionTime, lastTransactionTime));
+        }
+#endif // OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    }
+
+    if (!mle.IsFullThreadDevice() && mle.GetParent().IsThreadVersion1p1())
+    {
+        uint8_t pbbrServiceId;
+
+        SuccessOrExit(error = Get<BackboneRouter::Leader>().GetServiceId(pbbrServiceId));
+        SuccessOrExit(error = mle.GetServiceAloc(pbbrServiceId, messageInfo.GetPeerAddr()));
+    }
+    else
+    {
+        messageInfo.GetPeerAddr().SetToRoutingLocator(mle.GetMeshLocalPrefix(),
+                                                      Get<BackboneRouter::Leader>().GetServer16());
+    }
+
+    messageInfo.SetPeerPort(kCoapUdpPort);
+    messageInfo.SetSockAddr(Get<Mle::MleRouter>().GetMeshLocal16());
+
+    SuccessOrExit(error = Get<Coap::Coap>().SendMessage(*message, messageInfo, &DuaManager::HandleDuaResponse, this));
+
+    mIsDuaPending = true;
+
+    // TODO: (DUA) need update when CSL is enabled.
+    if (!Get<Mle::Mle>().IsRxOnWhenIdle())
+    {
+        Get<DataPollSender>().SendFastPolls();
+    }
+
+    otLogInfoDua("Sent DUA.req for DUA %s From %s to %s", dua.ToString().AsCString(),
+                 messageInfo.GetSockAddr().ToString().AsCString(), messageInfo.GetPeerAddr().ToString().AsCString());
+
+exit:
+    if (error == OT_ERROR_NO_BUFS)
+    {
+        UpdateCheckDelay(Mle::kNoBufDelay);
+    }
+
+    if (error != OT_ERROR_NONE && message != nullptr)
+    {
+        message->Free();
+    }
+}
+
+void DuaManager::HandleDuaResponse(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo, otError aResult)
+{
+    OT_UNUSED_VARIABLE(aMessageInfo);
+
+    mIsDuaPending = false;
+
+    if (aResult == OT_ERROR_RESPONSE_TIMEOUT)
+    {
+        UpdateCheckDelay(Mle::KResponseTimeoutDelay);
+        ExitNow();
+    }
+
+    VerifyOrExit(aResult == OT_ERROR_NONE && aMessage.GetCode() == OT_COAP_CODE_CHANGED, OT_NOOP);
+
+    ProcessDuaResponse(aMessage);
+
+exit:
+    mRegistrationTask.Post();
```

Comment:
not necessary to post registration task if ` UpdateCheckDelay(Mle::KResponseTimeoutDelay);` was executed?